js/newArticle.js

```diff
@@ -19,59 +22,46 @@ $('#write').on('keyup', function(){
   articlePreview.authorUrl = $('#article-author-url').val();
   articlePreview.publishedOn = new Date();
   articlePreview.category = $('#article-category').val();
-  articlePreview.body = '<pre><code>' + (marked($('#article-body').val())) + '</code></pre>';
+  articlePreview.markdown = marked($('#article-body').val());
 
-  cheese.articleList = new Article(articlePreview);
+  articleList.newSubmission = new Article(articlePreview);
 
   var articleTemplateRun = function () {
-    var theTemplateScript = $('#article-template').html();
-    var theTemplate = Handlebars.compile(theTemplateScript);
-    var theCompiledTemplate = theTemplate(cheese.articleList);
-    $('#articlesPreview').html(theCompiledTemplate);
+  $.get('../template/template.handlebars',function(data){
+    theTemplate = Handlebars.compile(data);
+    }).done(function(){
+      newPost.articleList = theTemplate(articleList.newSubmission);
+      $('#articlesPreview').html(newPost.articleList);
+    });
   };
```

Comment:
This event listener is doing a lot of things. It would probably benefit from refactoring into smaller functions. One problem is that every single keyup event is making an AJAX call to read the template.handlebars. That should probably only need to be done once and then stored in memory. Also `articleList.newSubmission = new Article(articlePreview);` is happening on every key press. Again, you should only need that to be created once.